### PR TITLE
Add missing require in eval mode for stn

### LINKED ISSUE #8: Eval model
Hi, thanks for this code and for the readme which explains very well how to use, even for someone completely new to Torch like me (I'm used to Caffe)

I trained the model using the following command line:
`luajit main.lua -n -1 --st --net idsia_net.lua --cnn 150,200,300,350 --locnet 200,300,200 --locnet3 150,150,150 -e 26 -o snapshot-train-2st-idsia`

It created two files in the current directory:
- snapshot-train-2st-idsia
- snapshot-train-2st-idsianorm

Now I would like to reload those weights to finetune, test the acuracy, etc. I guess I'm supposed to use `--eval` for this. So I run:
`luajit main.lua -n -1 --st --net idsia_net.lua --cnn 150,200,300,350 --locnet 200,300,200 --locnet3 150,150,150 -e 26 -o snapshot-train-2st-idsia --eval`

But **I get the following error:**

```
Loading training data...
Performing global normalization...
Performing local normalization...
Loading network from snapshot-train-2st-idsia
luajit: /root/torch/install/share/lua/5.1/torch/File.lua:343: unknown Torch class <nn.AffineTransformMatrixGenerator>
stack traceback:
    [C]: in function 'error'
    /root/torch/install/share/lua/5.1/torch/File.lua:343: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
    /root/torch/install/share/lua/5.1/nn/Module.lua:158: in function 'read'
    /root/torch/install/share/lua/5.1/torch/File.lua:351: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
    /root/torch/install/share/lua/5.1/nn/Module.lua:158: in function 'read'
    /root/torch/install/share/lua/5.1/torch/File.lua:351: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
    /root/torch/install/share/lua/5.1/nn/Module.lua:158: in function 'read'
    /root/torch/install/share/lua/5.1/torch/File.lua:351: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:369: in function 'readObject'
    /root/torch/install/share/lua/5.1/nn/Module.lua:158: in function 'read'
    /root/torch/install/share/lua/5.1/torch/File.lua:351: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:409: in function 'load'
    main.lua:81: in main chunk
    [C]: at 0x00406670
```

And line 81 corresponds to:

``` lua
local network
if opt.eval then
  if opt.output then
    print('Loading network from '..opt.output)
    network = torch.load(opt.output)  -- here, line 81
  else
    error('Must supply the network to use in eval mode')
  end
```

### PR DESCRIPTION
**Pull request by albanD**
Fix #8

The only drawback is that when using the `--eval` mode with a network containing a Spatial Transformer, the `--st` must also be specified otherwise the network will not be loaded properly. This could be prevented by doing a `pcall` on the `torch.load` and retry it after requirering `stn` but it does not feel like the right thing to do.

### CODE REVIEW

**Comment by deltheil** [comment]
Thanks!
